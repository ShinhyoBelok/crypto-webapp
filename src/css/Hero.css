.hero {
  align-items: center;
  justify-content: center;
  gap: 1rem;
  padding: 1rem;
}
.hero img {
  width: 8rem;
  filter: invert(34%) sepia(29%) saturate(1460%) hue-rotate(287deg) brightness(90%) contrast(97%);
}