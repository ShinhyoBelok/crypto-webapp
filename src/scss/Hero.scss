.hero {
  align-items: center;
  justify-content: center;
  gap: 15%;
  padding: 1rem;
  img {
    width: 8rem;
    filter: invert(34%) sepia(29%) saturate(1460%)  hue-rotate(287deg) brightness(90%) contrast(97%);
  }
  h5 {
    font-weight: 300;
  }
}
