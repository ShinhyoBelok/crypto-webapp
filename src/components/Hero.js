import { string } from 'prop-types';
import React from 'react';
import '../css/Hero.css';

export default function Hero(props) {
  const {
    name,
    rank,
  } = props;

  return (
    <section className="hero flex">
      <img src="https://cdn-icons-png.flaticon.com/512/3461/3461335.png" alt="Trading Graph" />
      <div>
        <h2>{name}</h2>
        <h5>
          TOP&ensp;
          {rank}
        </h5>
      </div>
    </section>
  );
}

Hero.propTypes = {
  name: string,
  rank: string,
};

Hero.defaultProps = {
  name: 'CRYPTOS',
  rank: '100',
};
