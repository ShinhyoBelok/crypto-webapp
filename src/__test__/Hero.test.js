import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Hero from '../components/Hero';

describe('Hero component', () => {
  test('Test Hero button render correctly', () => {
    const hero = render(
      <Provider store={store}>
        <BrowserRouter>
          <Hero />
        </BrowserRouter>
      </Provider>,
    );
    expect(hero).toMatchSnapshot();
  });
});
