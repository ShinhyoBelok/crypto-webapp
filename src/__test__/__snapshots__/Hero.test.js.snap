// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component Test Hero button render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="hero flex"
      >
        <img
          alt="Trading Graph"
          src="https://cdn-icons-png.flaticon.com/512/3461/3461335.png"
        />
        <div>
          <h2>
            CRYPTOS
          </h2>
          <h5>
            TOP 
            100
          </h5>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="hero flex"
    >
      <img
        alt="Trading Graph"
        src="https://cdn-icons-png.flaticon.com/512/3461/3461335.png"
      />
      <div>
        <h2>
          CRYPTOS
        </h2>
        <h5>
          TOP 
          100
        </h5>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
